xbmc/cores/VideoPlayer/test/edl/TestEdl.cpp

```diff
@@ -47,9 +48,9 @@ TEST_F(TestEdl, TestParsingMplayerTimeBasedEDL)
   // file has only 1 cut starting at 5.3 secs and ending at 7.1 secs
   // so total cut time should be 1.8 seconds (1800 msec)
   EXPECT_EQ(edl.HasCuts(), true);
-  EXPECT_EQ(edl.GetTotalCutTime(), 1.8 * 1000);
+  EXPECT_TRUE(FloatEquals(edl.GetTotalCutTime(), 1.8 * 1000, 1));
   EXPECT_EQ(edl.GetCutMarkers().size(), 1);
-  EXPECT_EQ(edl.GetCutMarkers().at(0), 5.3 * 1000); // 5.3 secs
+  EXPECT_TRUE(FloatEquals(edl.GetCutMarkers().at(0), 5.3 * 1000, 1)); // 5.3 secs
```

Comment:
`FloatEquals` is only for compare floats or doubles avoiding rounding errors....  threshold should be 0.001, 0.00001 etc. not "1" integer numbers. 

Something is very wrong here :)

Reply:
That's I actually where I don't understand how to fix it properly… Under the hood, there are rounding operations in these functions, and the tests show that 5.2 to 5.3 multiplied by 1000 grows everything out of control

Reply:
I decided to check what's wrong with the test I mentioned and crafted the following minimal reproducer:

```c++
#include <cmath>
#include <cstdio>

int main()
{
        printf("%lld\n", static_cast<int64_t>(atof("5.3") * 1000));
        return 0;
}
```

```shell
$ g++ -m64 -o test64 test.cpp && g++ -m32 -o test32 test.cpp && echo '32:' && ./test32 && echo "" && echo '64:' && ./test64
32:
5299

64:
5300
```